crates/node/src/commands/run.rs

```diff
@@ -2,15 +2,48 @@ use std::path::PathBuf;
 
 use madara_runtime::SealingMode;
 use mc_data_availability::DaLayer;
-use mc_l1_messages::config::L1MessagesWorkerConfig;
-use mc_l1_messages::error::L1MessagesConfigError;
+use mc_l1_messages::config::{L1MessagesWorkerConfig, L1MessagesWorkerConfigError};
 use sc_cli::{Result, RpcMethods, RunCmd, SubstrateCli};
 use sc_service::BasePath;
 use serde::{Deserialize, Serialize};
 
 use crate::cli::Cli;
 use crate::service;
 
+#[derive(Debug, Clone, clap::Args)]
+pub struct L1MessagesParams {
+    /// Ethereum Provider (Node) Url
+    #[clap(
+        long,
+        requires="l1_contract_address",
+        value_hint=clap::ValueHint::Url,
+    )]
+    pub provider_url: String,
+
+    /// L1 Contract Address
+    #[clap(
+        long,
+        requires="provider_url",
+        value_hint=clap::ValueHint::Other,
+    )]
+    pub l1_contract_address: String,
+}
+
+#[derive(Debug, Clone, clap::Args)]
+pub struct L1Messages {
+    /// Path to configuration file for Ethereum Core Contract Events Listener
+    #[clap(
+        long,
+        conflicts_with="provider_url",
+        conflicts_with="l1_contract_address",
+        value_hint=clap::ValueHint::FilePath,
+    )]
+    pub l1_messages_config: Option<PathBuf>,
+
+    #[clap(flatten)]
+    pub config_params: Option<L1MessagesParams>,
```

Comment:
Here you can probably state that this one conflict with `l1_messages_config`

Reply:
```sh
error: methods are not allowed for flattened entry
  --> crates/node/src/commands/run.rs:41:12
   |
41 |     #[clap(flatten, conflicts_with="l1_messages_config")]
```

Reply:
and if you don't flatten?

Reply:
what we want express is that only one of the two should be used.
Maybe the good way is to put them in an arg group and there is some "only one in those" syntax

Reply:
without `flatten`
```sh
error[E0599]: the method `value_parser` exists for reference `&&&&&&_AutoValueParser<L1MessagesParams>`, but its trait bounds were not satisfied
    --> crates/node/src/commands/run.rs:41:5
     |
14   |   pub struct L1MessagesParams {
     |   ---------------------------
     |   |
     |   doesn't satisfy `L1MessagesParams: From<&'s std::ffi::OsStr>`
     |   doesn't satisfy `L1MessagesParams: From<&'s str>`
     |   doesn't satisfy `L1MessagesParams: From<OsString>`
     |   doesn't satisfy `L1MessagesParams: From<std::string::String>`
     |   doesn't satisfy `L1MessagesParams: FromStr`
     |   doesn't satisfy `L1MessagesParams: ValueEnum`
     |   doesn't satisfy `L1MessagesParams: ValueParserFactory`
...
41   | /     #[clap(conflicts_with="l1_messages_config")]
42   | |     pub config_params: Option<L1MessagesParams>,
     | |_______________________________________________^ method cannot be called on `&&&&&&_AutoValueParser<L1MessagesParams>` due to unsatisfied trait bounds
     |
    ::: /Users/danielbigos/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_builder-4.4.8/src/builder/value_parser.rs:2462:1
     |
2462 |   pub struct _AutoValueParser<T>(std::marker::PhantomData<T>);
     |   ------------------------------ doesn't satisfy `_: _ValueParserViaParse`
     |
     = note: the following trait bounds were not satisfied:
             `L1MessagesParams: ValueEnum`
             which is required by `&&&&&_AutoValueParser<L1MessagesParams>: clap::builder::via_prelude::_ValueParserViaValueEnum`
             `L1MessagesParams: ValueParserFactory`
             which is required by `&&&&&&_AutoValueParser<L1MessagesParams>: clap::builder::via_prelude::_ValueParserViaFactory`
             `L1MessagesParams: From<OsString>`
             which is required by `&&&&_AutoValueParser<L1MessagesParams>: clap::builder::via_prelude::_ValueParserViaFromOsString`
             `L1MessagesParams: From<&'s std::ffi::OsStr>`
             which is required by `&&&_AutoValueParser<L1MessagesParams>: clap::builder::via_prelude::_ValueParserViaFromOsStr`
             `L1MessagesParams: From<std::string::String>`
             which is required by `&&_AutoValueParser<L1MessagesParams>: clap::builder::via_prelude::_ValueParserViaFromString`
             `L1MessagesParams: From<&'s str>`
             which is required by `&_AutoValueParser<L1MessagesParams>: clap::builder::via_prelude::_ValueParserViaFromStr`
             `L1MessagesParams: FromStr`
             which is required by `_AutoValueParser<L1MessagesParams>: clap::builder::via_prelude::_ValueParserViaParse`
note: the traits `ValueEnum`, `ValueParserFactory`, `From`,  and `FromStr` must be implemented
    --> /Users/danielbigos/.rustup/toolchains/nightly-2023-08-24-x86_64-apple-darwin/lib/rustlib/src/rust/library/core/src/str/traits.rs:603:1
     |
603  | pub trait FromStr: Sized {
     | ^^^^^^^^^^^^^^^^^^^^^^^^
     |
    ::: /Users/danielbigos/.rustup/toolchains/nightly-2023-08-24-x86_64-apple-darwin/lib/rustlib/src/rust/library/core/src/convert/mod.rs:538:1
     |
538  | pub trait From<T>: Sized {
     | ^^^^^^^^^^^^^^^^^^^^^^^^
     |
    ::: /Users/danielbigos/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_builder-4.4.8/src/derive.rs:264:1
     |
264  | pub trait ValueEnum: Sized + Clone {
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     |
    ::: /Users/danielbigos/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_builder-4.4.8/src/builder/value_parser.rs:2312:1
     |
2312 | pub trait ValueParserFactory {
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     = note: this error originates in the macro `clap::value_parser` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0599`.
error: could not compile `madara` (bin "madara") due to 2 previous errors
```